src/components/molecules/PlanRoom/RoomTables/Table/index.tsx

```diff
@@ -0,0 +1,182 @@
+import { useCallback, useState, SetStateAction } from 'react';
+import { Box, Textarea } from '@chakra-ui/react';
+import update from 'immutability-helper';
+import { useDrop, XYCoord, useDrag } from 'react-dnd';
+import ItemTypes from 'components/molecules/PlanRoom/RoomTables/TableDNDTypes';
+import { TypeItem, TableType, TablesType } from 'components/molecules/PlanRoom/RoomTables/TableComponentsType';
+import SaveBoard from 'components/molecules/PlanRoom/SaveRoomTables/SaveBoard';
+import { theme } from 'others/theme';
+
+let changeValueTable: Function;
+let changeValueTableBorderRadius: Function;
+let addNewTable: Function;
+let save: Function;
+
+const Items = () => {
+  const [tables, setTables] = useState<TablesType>({
+    0: { top: 0, left: 0, text: '0', width: 100, height: 50, borderRadius: 0 },
+  });
+
+  const moveTable = useCallback(
+    (id: string, left: number, top: number) => {
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { left, top },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  changeValueTable = useCallback(
+    (id: string, text: string) => {
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { text },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  changeValueTableBorderRadius = useCallback(
+    (id: string, borderRadius: number) => {
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { borderRadius },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  addNewTable = (id: number) => {
+    tables[id] = { top: 0, left: 0, text: id.toString(), width: 100, height: 50, borderRadius: 0 };
+    let top = Math.floor(Math.random() * 100) + 40;
```

Comment:
I guess the reason for this was to not stack new tables, but because you are passing previous index, you move randomly previously added table.

Reply:
The issue with the index will be corrected